sound/soc/sof/ipc4-mtrace.c

```diff
@@ -424,9 +424,8 @@ static int ipc4_mtrace_enable(struct snd_sof_dev *sdev)
 	msg.primary |= SOF_IPC4_MOD_INSTANCE(SOF_IPC4_MOD_INIT_BASEFW_INSTANCE_ID);
 	msg.extension = SOF_IPC4_MOD_EXT_MSG_PARAM_ID(SOF_IPC4_FW_PARAM_SYSTEM_TIME);
 
-	/* The system time is in usec, UTC, epoch is 1601-01-01 00:00:00 */
-	kt = ktime_add_us(ktime_get_real(), FW_EPOCH_DELTA * USEC_PER_SEC);
-	system_time = ktime_to_us(kt);
+	/* convert ns to us since system time is in us format */
+	system_time = local_clock() / NSEC_PER_USEC;
```

Comment:
maybe a stupid question but why not use ktime_get() and ktime_to_us() ?

Reply:
@plbossart  It was my first try, but later I found the time was different with the time stamp used by dmesg, so it is not helpful for debug. I debugged the dmesg tools and finally found it use local_clock, so I use the same time api to work in the same time domain.

Reply:
this is a test result with the following patch. The difference is about 8 ms, and we will find the fw timestamp is about 8ms ahead kernel log.

[   80.690381] sof-audio-pci-intel-mtl 0000:00:1f.3: ----------- system 80.682282

````c
+       kt = ktime_get();
+       system_time = ktime_to_us(kt);
 
+       dev_info(sdev->dev, "----------- system %lld", system_time);
````

with local_clock, the difference is all most zero 
[   27.907670] sof-audio-pci-intel-mtl 0000:00:1f.3: ++++++++ system 27.907670

Reply:
probably because of some NTP time correction?

At any rate it'd be good to add a comment that local_clock() is used to align with dmesg, so both kernel and firmware logs have the same base and a minor delta due to the IPC.

Reply:
> > but later I found the time was different with the time stamp used by dmesg, so it is not helpful for debug. I debugged the dmesg tools and finally found it use local_clock, so I use the same time api to work in the same time domain.

> At any rate it'd be good to add a comment that local_clock() is used to align with dmesg

These differences should be documented:
https://www.kernel.org/doc/html/latest/timers/timekeeping.html

If that 8ms difference can be explained by this page, then use its terminology in a comment and refer to it.

If it cannot be explained by this page, then this PR is "programming by coincidence".

Reply:
Thanks Google,  [cpuidle: Change ktime_get() with local_clock()](https://patchwork.kernel.org/project/linux-pm/patch/1461228990-28957-1-git-send-email-daniel.lezcano@linaro.org/) This patch shows that the ktime_get() can have a non negligeable overhead, use local_clock() instead.  changing the ktime_get() by the local_clock() allows to save 80ns at idle time (entry + exit). And in some circumstances, especially when there are several CPUs racing for the clock access, we save tens of microseconds. Maybe this is why system log use local_clock()